### PR TITLE
Fix #2888 - Multi-line errors in ls mess with tab writing

### LINKED ISSUE #2888: Multi-line errors in ls mess with tab writing
I have an `amazonec2` machine in invalid state which seems to return multi-line error message:

```console
$ dm ls
NAME      ACTIVE   URL         STATE   URL   SWARM   DOCKER    ERRORS
debtest   -        amazonec2   Error                 Unknown   MissingParameter: The request must contain the parameter InstanceId
          status code: 400, request id:
default   *    virtualbox   Running   tcp://192.168.99.100:2376        v1.9.1
rc        -    virtualbox   Running   tcp://192.168.99.103:2376        v1.10.0-rc1
```

Maybe we can figure out a good way to accomodate them, or at least strip newlines out of errors which appear in the `ERRORS` column.

cc @docker/machine-maintainers

### PR DESCRIPTION
**Pull request by jeanlaurent**

### CODE REVIEW

File: commands/ls.go [hunk 1]
```diff
@@ -468,18 +468,22 @@ func getHostListItems(hostList []*host.Host, hostsInError map[string]error) []Ho
...
-		hostListItems = append(hostListItems, itemInError)
+		hostListItems = append(hostListItems, newHostListItemInError(name, err))
 	}
 
 	sortHostListItemsByName(hostListItems)
 	return hostListItems
 }
 
+func newHostListItemInError(name string, err error) HostListItem {
+	itemInError := HostListItem{}
```
Comment (nathanleclaire) [inline commands/ls.go]: Why not put these properties directly in the literal, i.e.:

```go
return HostListItem{
    Name: name,
    DriverName: "not found",
    State: state.Errror,
    Error: strings.Replace(err.Error(), "\n", " ", -1)
}
```

**Comment by nathanleclaire** [comment]
I have one nitpick and there seems to be a build error but once those are fixed LGTM

**Comment by jeanlaurent** [comment]
Thx for your input, ( old java way of doing things, still kick'in some time...)

This is fixed.

@docker/machine-maintainers PTAL

**Comment by dgageot** [comment]
LGTM

**Comment by nathanleclaire** [comment]
> And I should not perform push request while at a conference I believe...

Heh -- only if the WiFi is good!

**Pull request merged**